### ISSUE TITLE
[Config Support]: Docker compose Ubuntu 22.04 install help (for noob)

### ISSUE THREAD

**Issue by joekingmo** [blakeblackshear/frigate#3627]
### Describe the problem you are having

Hello, 
I have a situation that I've been struggling with, and could use some help. I suspect it is a simple solution for someone smarter than me. I'm trying to move my HassOS Frigate Add-on to run on my host machine. Everything works great in HA, but falls apart when I try to load it on the host. I'm looking to have more control over the storage and gain any efficiencies by not virtualizing the instance. 

I've tried to find the solution on my own, but I'm failing to get any real progress. Hoping someone can help me out here. 

Bottom line: my docker version of frigate will not load the webpage (header with icon visible, spinning wheel in main section) while my HassOS add-on works great

Any help would be much appreicated!

**Host Server:**
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy

**Docker Version:**
Client: Docker Engine - Community
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:02:46 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.17
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       a89b842
  Built:            Mon Jun  6 23:00:51 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.7
  GitCommit:        0197261a30bf81f1ee8e6a4dd2dea0ef95d67ccb
 runc:
  Version:          1.1.3
  GitCommit:        v1.1.3-0-g6724737
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

### Version

won't load

### Frigate config file

```yaml
mqtt:
  host: ****
  user: ****
  password: ****
  topic_prefix: frigate
  client_id: frigate

detectors:
  coral:
    type: edgetpu
    device: usb

cameras:
  driveway: 
    ffmpeg:
      inputs:
        - path: rtsp://****:****@192.168.1.60:554/Streaming/Channels/2
          roles:
            - detect
            - rtmp
        - path: rtsp://****:****@192.168.1.60:554/Live
          roles:
            - record
    rtmp:
      enabled: True 
    detect:
      width: 640
      height: 360
  porch: 
    ffmpeg:
      inputs:
        - path: rtsp://****:****@192.168.1.20:554/s0
          roles:
            - detect
            - rtmp
        - path: rtsp://****:****@192.168.1.20:554/s0
          roles:
            - record
    rtmp:
      enabled: True 
    detect:
      width: 1920
      height: 1080

objects:
  track:
    - person
    - car
    - motorcycle

record:
  enabled: True
  events:
    retain:
      default: 21
      mode: active_objects
```

### Relevant log output

```shell
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-08-10 16:32:14] frigate.app                    INFO    : Starting Frigate (0.10.1-83481af)
[2022-08-10 16:32:14] frigate.app                    INFO    : Creating directory: /tmp/cache
Starting migrations
[2022-08-10 16:32:14] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-08-10 16:32:14] peewee_migrate                 INFO    : There is nothing to migrate
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

USB

**Comment by NickM-27** [comment]
There must be more logs than this, for example the logs saying the USB TPU was found. Likely there will also be an error in the logs showing what happened

**Comment by NickM-27** [comment]
Also please include your docker-compose.yaml

**Comment by joekingmo** [comment]
> Also please include your docker-compose.yaml

Sorry about that. Here are the contents of docker-compose.yaml file.

```yaml
version: "3.9"
services:
  frigate:
    container_name: frigate
    privileged: true # this may not be necessary for all setups
    restart: unless-stopped
    image: blakeblackshear/frigate:stable-amd64
    shm_size: "128mb" # update for your cameras based on calculation above
    devices:
      - /dev/bus/usb:/dev/bus/usb # passes the USB Coral, needs to be modified for other versions
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ./config/config.yml:/config/config.yml:ro
      - /mnt/shares/video:/media/frigate
    ports:
      - "5000:5000"
      - "1935:1935" # RTMP feeds
    environment:
      FRIGATE_RTSP_PASSWORD: "***"
```

**Comment by NickM-27** [comment]
You need to add `:rw` at the end of your media mount otherwise frigate won't have write permission to be able to update the database and start

**Comment by joekingmo** [comment]
> You need to add `:rw` at the end of your media mount otherwise frigate won't have write permission to be able to update the database and start

Updated the line to ` - /mnt/shares/video:/media/frigate:rw`. Same behavior. This was after stopping the container and doing `docker system prune` just to be sure.

**Comment by NickM-27** [comment]
Need to see the full logs then

**Comment by joekingmo** [comment]
This is the output of `docker logs -f frigate`, I presume that is what you are after. If not, please let me know where I can find them. 

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-08-10 18:40:24] frigate.app                    INFO    : Starting Frigate (0.10.1-83481af)
[2022-08-10 18:40:24] frigate.app                    INFO    : Creating directory: /tmp/cache
Starting migrations
[2022-08-10 18:40:24] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-08-10 18:40:24] peewee_migrate                 INFO    : There is nothing to migrate
timed out
[cmd] python3 exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-08-10 18:41:31] frigate.app                    INFO    : Starting Frigate (0.10.1-83481af)
Starting migrations
[2022-08-10 18:41:31] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-08-10 18:41:31] peewee_migrate                 INFO    : There is nothing to migrate
timed out
[cmd] python3 exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-08-10 18:42:37] frigate.app                    INFO    : Starting Frigate (0.10.1-83481af)
Starting migrations
[2022-08-10 18:42:37] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-08-10 18:42:37] peewee_migrate                 INFO    : There is nothing to migrate
timed out
[cmd] python3 exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-08-10 18:43:43] frigate.app                    INFO    : Starting Frigate (0.10.1-83481af)
Starting migrations
[2022-08-10 18:43:43] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-08-10 18:43:43] peewee_migrate                 INFO    : There is nothing to migrate
timed out
[cmd] python3 exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-08-10 18:44:49] frigate.app                    INFO    : Starting Frigate (0.10.1-83481af)
Starting migrations
[2022-08-10 18:44:49] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-08-10 18:44:49] peewee_migrate                 INFO    : There is nothing to migrate
timed out
[cmd] python3 exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[2022-08-10 18:45:55] frigate.app                    INFO    : Starting Frigate (0.10.1-83481af)
Starting migrations
[2022-08-10 18:45:55] peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
[2022-08-10 18:45:55] peewee_migrate                 INFO    : There is nothing to migrate

**Comment by joekingmo** [comment]
@NickM-27 , I'm up and running now. You can disregard. As it turns out, the issue was with the docker network. `/etc/docker/daemon.json` had `br0` defined. Once I removed that entry, everything came up as expected. I do appreciate you trying to help me out.

**Comment by joekingmo** [comment]
Just in case anyone else runs into the same issue I had. I ended up having to add this line into docker-compose.yaml `network_mode: "host"`. I understand this issue to be related to Ubuntu not playing nice with docker networks and bridge used by the hypervisor. This [article](https://wiki.archlinux.org/title/Docker#Starting_Docker_breaks_KVM_bridged_networking) covers the changes I had to make to `/etc/docker/daemon.json` (shown below).

The below configuration is required for my VMs to work. Without it, my VMs broke, but the frigate container worked. So, it was an either/or scenario. Which obviously didn't work for me. Adding in the `network_mode` key in my docker compose file **combined** with the docker config below, made it to where both docker and my VMs worked simataneously.... finally. 
`
{
  "bridge": "br0"
}
`
